src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -2038,13 +2038,14 @@ public static void ValueType_Properties_NotIgnoredWhen_IgnoreNullValues_Active_C
             var options = new JsonSerializerOptions { IgnoreNullValues = true };
 
             // Deserialization.
-            string json = @"{""MyString"":null,""MyInt"":0,""MyPointClass"":null,""MyPointStruct"":{""X"":0,""Y"":0}}";
+            string json = @"{""MyString"":null,""MyInt"":0,""MyBool"":null,""MyPointClass"":null,""MyPointStruct"":{""X"":0,""Y"":0}}";
```

Comment:
This and related changes are just to add coverage for nullable structs, as a follow up to https://github.com/dotnet/runtime/pull/39147.

The main test changes to review are `Ignore_WhenWriting_Default_Globally`, `Ignore_WhenWriting_Default_PerPropoerty`, and `JsonIgnoreCondition_WhenWritingNull_OnValueType_Fail` below.